plugin.rb

```diff
@@ -16,6 +16,47 @@ class CrowdAuthenticatorMode
   def after_create_account(user, auth)
   end
 
+  def set_groups(user, auth)
+    return unless SiteSetting.crowd_groups_enabled
+    Rails.logger.debug("DEBUG: auth: '#{auth.inspect}'")
+    user_crowd_groups = ( auth[:info] && auth[:info].groups ) ? auth[:info].groups : nil
+    group_map = Hash.new
+    check_groups = Hash.new
+    SiteSetting.crowd_groups_mapping.split("|").each { |map|
+      keyval = map.split(":", 2)
+      group_map[keyval[0]] = keyval[1]
+      check_groups[keyval[1]] = 0
+    }
+    if !(user_crowd_groups == nil || group_map.empty?)
+      user_crowd_groups.each { |user_crowd_group|
```

Comment:
Did you run this through our linting tools (rubocop?) I thought we didn't allow multi-line blocks without `do`.

Reply:
I didn't find any instructions about rubocop (and apologies I'm completely new to ruby), any pointers would be welcome.
At some point in the future I need to write test for another fix so it will be handy then too.